lib/auth/auth_login_form.dart

```diff
@@ -0,0 +1,100 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<AuthBloc, AuthState>(
+      listener: (context, state) {
+        if (state is AuthSignInSuccess) {
+          Navigator.of(context).pushNamed('/');
+        }
+      },
+      builder: (context, state) {
+        final bool isAuthSignInFailure = state is AuthSignInFailure;
+        return Padding(
+            padding: EdgeInsets.all(16),
+            child: FormBuilder(
+                autovalidate: _autoValidate,
+                key: _fbKey,
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: <Widget>[
+                    FormBuilderTextField(
+                      attribute: "email",
+                      decoration: InputDecoration(
+                        icon: Icon(Icons.email),
+                        labelText: "Email",
+                        errorText: isAuthSignInFailure ? "" : null,
+                      ),
+                      validators: [
+//                        FormBuilderValidators.required(errorText: "Required"),
+//                        FormBuilderValidators.email(errorText: "Invalid")
+                      ],
+                    ),
+                    FormBuilderTextField(
+                      attribute: "password",
+                      decoration: InputDecoration(
+                        icon: Icon(Icons.lock),
+                        labelText: "Password",
+                        errorText: isAuthSignInFailure ? "" : null,
+                      ),
+                      validators: [
+                        FormBuilderValidators.required(errorText: "Required"),
+                      ],
+                    ),
+                    Container(
+                      margin: EdgeInsets.only(top: 30),
+                      child: state is AuthSignInFailure
+                          ? _getErrorWidget(state.error)
+                          : null,
+                    ),
+                    Padding(
+                      padding: EdgeInsets.only(top: 16),
+                      child: RaisedButton(
+                        color: Theme.of(context).buttonColor,
+                        onPressed: () => _validateInputs(),
+                        child: Text("Log in"),
+                      ),
+                    ),
+                  ],
+                )));
+      },
+    );
+  }
+
+  Widget _getErrorWidget(String text) {
+    return Text(
+      text,
+      style: TextStyle(fontWeight: FontWeight.bold, color: Colors.redAccent),
+    );
+  }
+
+  void _validateInputs() {
+    if (_fbKey.currentState.validate()) {
+      _fbKey.currentState.save();
+      BlocProvider.of<AuthBloc>(context).add(AuthSignedIn(
+          body: AuthModal(
+              email: _fbKey.currentState.value["email"],
+              password: _fbKey.currentState.value["password"])));
```

Comment:
I think we need some code style. I offer to use more tree code structures. The best way to apply it is to put commas after the right brackets.

Reply:
Fix